File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.py
from collections import *
import sys
sys.setrecursionlimit(10000)
def input():return sys.stdin.readline().rstrip()
N,M,V=map(int,input().split())
g=[[] for _ in range(N+1)]
visited=[0 for _ in range(N+1)]
for i in range(M):
    a,b=map(int,input().split())
    g[a].append(b)
    g[b].append(a)
def dfs(cur):
    print(cur,end=' ')
    visited[cur]=1
    for nxt in g[cur]:
        if visited[nxt]:continue
        dfs(nxt)
def bfs(start):
    q=deque()
    q.append(start)
    visited[start]=1
    while q:
        cur=q.popleft()
        print(cur,end=' ')
        for nxt in g[cur]:
            if visited[nxt]:continue
            visited[nxt]=1
            q.append(nxt)
for i in range(1,N+1):
    g[i].sort()
dfs(V)
visited=[0 for _ in range(N+1)]
print('')
bfs(V)

File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.java
import java.io.*;
import java.util.*;

public class Main {

    static int[][] arr;
    static boolean visited[];
    static int N;
    public static void bfs(int start){
        Queue<Integer> q= new LinkedList<>();
        q.add(start);
        visited[start]=true;
        while (!q.isEmpty())
        {
            int cur=q.poll();
            System.out.printf("%d ",cur);
            for (int i=1;i<=N;i++)
            {
                if (arr[cur][i]==1 && visited[i]==false){
                    visited[i]=true;
                    q.add(i);
                }
            }
        }

    }
    public static void dfs(int cur){
        visited[cur]=true;
        System.out.printf("%d ",cur);
        for (int i=1;i<=N;i++)
        {
            if (arr[cur][i]==1 && visited[i]==false)
            {
                dfs(i);
            }
        }
    }
    public static void main(String[] args) throws IOException {
        BufferedReader br= new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st= new StringTokenizer(br.readLine());
        int M,V;
        N=Integer.parseInt(st.nextToken());
        M=Integer.parseInt(st.nextToken());
        V=Integer.parseInt(st.nextToken());
        arr=new int[N+1][N+1];

        for (int i=0;i<M;i++)
        {
            StringTokenizer str= new StringTokenizer(br.readLine());
            int q,w;
            q=Integer.parseInt(str.nextToken());
            w=Integer.parseInt(str.nextToken());
            arr[q][w]=1;
            arr[w][q]=1;
        }
        visited=new boolean[N+1];
        dfs(V);
        System.out.println();
        visited=new boolean[N+1];
        bfs(V);
    }
}